cpu/stm32/kconfigs/Kconfig.clk

```diff
@@ -112,19 +112,23 @@ endif  # CPU_FAM_G4 || CPU_FAM_L4 || CPU_FAM_L5
 
 endif  # CPU_FAM_G0 || CPU_FAM_G4 || CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB
 
-if CPU_FAM_F0
+if CPU_FAM_F0 || CPU_FAM_F1 || CPU_FAM_F3
 config CLOCK_PLL_PREDIV
-    int "PLLIN division factor" if CUSTOM_PLL_PARAMS && !CPU_LINE_STM32F031X6 && !CPU_LINE_STM32F042X6
-    default 2 if CPU_LINE_STM32F031X6 || CPU_LINE_STM32F042X6
+    int "PLLIN division factor" if CUSTOM_PLL_PARAMS && !CPU_LINE_STM32F031X6 && !CPU_LINE_STM32F042X6 && !CPU_LINE_STM32F303X8
+    # iotlab based boards provide a 16MHz HSE so they need a predivider of 2
+    # to remain with a 72MHz sysclk by default.
+    default 2 if CPU_LINE_STM32F031X6 || CPU_LINE_STM32F042X6 || CPU_LINE_STM32F303X8 || (CPU_FAM_F1 && (BOARD_IOTLAB_M3 || BOARD_IOTLAB_A8_M3 || BOARD_FOX))
```

Comment:
Do you think it is possible to generalize this with a feature instead of specifying the boards here?

Reply:
What kind of feature could you imagine ?
For f0/f3, we could define a symbol `CLOCK_PLL_NO_PREDIV` that is false by default and select it in the CPU families that have this limitation. And here that would just be `default 2 if CLOCK_PLL_NO_PREDIV`.
For the iotlab boards case, it's more complicated since it depends on the HSE clock frequency. I have no idea how to solve this cleanly. With f2/f4/f7, this will become even more difficult to manage since there are more HSE values (12MHz, 16MHz, 25MHz, etc)

Reply:
What about placing `board.config` files for the boards to set their default then?

Reply:
Are they used in `make menuconfig` afterwards ? If yes, that would be a good solution to that mess.

Reply:
Yes, but so far they have to be added to the list of configurations to load like here:
https://github.com/RIOT-OS/RIOT/blob/7237e384368a63e79b3b7273783b5230c29e448b/cpu/cortexm_common/Makefile.features#L41-L45

In this case it would be always added so no need to check for `TEST_KCONFIG`.

Reply:
Did that in dc9e525 but then menuconfig fails:

```
$ make BOARD=iotlab-m3 -C examples/hello-world menuconfig --no-print-directory 
[genconfig.py]:ERROR-=> The parameter CLOCK_PLL_PREDIV could not be set to 2.
Traceback (most recent call last):
  File "/work/riot/RIOT/dist/tools/kconfiglib/genconfig.py", line 440, in <module>
    main()
  File "/work/riot/RIOT/dist/tools/kconfiglib/genconfig.py", line 394, in main
    if not check_configs(kconf) and not args.ignore_config_errors:
  File "/work/riot/RIOT/dist/tools/kconfiglib/genconfig.py", line 251, in check_configs
    sym_check = check_config_symbols(kconf)
  File "/work/riot/RIOT/dist/tools/kconfiglib/genconfig.py", line 171, in check_config_symbols
    rng = get_sym_applying_range(sym)
  File "/work/riot/RIOT/dist/tools/kconfiglib/genconfig.py", line 297, in get_sym_applying_range
    "(if {})".format(cond.name) if not cond.is_constant else "")
AttributeError: 'tuple' object has no attribute 'is_constant'
make: *** No rule to make target '/work/riot/RIOT/examples/hello-world/bin/iotlab-m3/generated/out.config', needed by 'menuconfig'.  Stop.
```

Any idea @leandrolanzieri ?

Reply:
> Did that in [dc9e525](https://github.com/RIOT-OS/RIOT/commit/dc9e525b1c3004ad2d4ac7c17a5229e01380d97e) but then menuconfig fails:
> 
> ```
> $ make BOARD=iotlab-m3 -C examples/hello-world menuconfig --no-print-directory 
> [genconfig.py]:ERROR-=> The parameter CLOCK_PLL_PREDIV could not be set to 2.
> Traceback (most recent call last):
>   File "/work/riot/RIOT/dist/tools/kconfiglib/genconfig.py", line 440, in <module>
>     main()
>   File "/work/riot/RIOT/dist/tools/kconfiglib/genconfig.py", line 394, in main
>     if not check_configs(kconf) and not args.ignore_config_errors:
>   File "/work/riot/RIOT/dist/tools/kconfiglib/genconfig.py", line 251, in check_configs
>     sym_check = check_config_symbols(kconf)
>   File "/work/riot/RIOT/dist/tools/kconfiglib/genconfig.py", line 171, in check_config_symbols
>     rng = get_sym_applying_range(sym)
>   File "/work/riot/RIOT/dist/tools/kconfiglib/genconfig.py", line 297, in get_sym_applying_range
>     "(if {})".format(cond.name) if not cond.is_constant else "")
> AttributeError: 'tuple' object has no attribute 'is_constant'
> make: *** No rule to make target '/work/riot/RIOT/examples/hello-world/bin/iotlab-m3/generated/out.config', needed by 'menuconfig'.  Stop.
> ```
> 
> Any idea @leandrolanzieri ?

The Kconfig error is due to trying to configure `CONFIG_CLOCK_PLL_PREDIV` without setting `CONFIG_CUSTOM_PLL_PARAMS=y`. Adding that to the config file should fix the issue.

The horrible python error seems to be a bug on my `genconfig.py` script. I'll take a look :)

Reply:
> Adding that to the config file should fix the issue.

Thanks, that was it!

Reply:
> The horrible python error seems to be a bug on my genconfig.py script. I'll take a look :)

See #15594